### PR TITLE
Closes #299: edit product and data entries

### LINKED ISSUE #299: Product/Data Entry inventory changes: edit entries
When viewing the Edit Inventory page, users need to be able to easily edit or make changes to their existing product and data entries.

* When user lands on Edit Inventory page, display all product or data entries with an edit icon next to their row.
* When clicked, display the existing product/data entry modal with all available information pre-populated in the appropriate fields.
* Allow users to make changes to the fields and click 'Done' to save their work.
Reflect the new changes and updates in the system.

### PR DESCRIPTION
**Pull request by juliosantos**
This can be tested by navigating to `/inventories/:inventory_id/edit`. You'll see the new **edit** button to the left of each row.

### CODE REVIEW

**Comment by MakotoTheKnight** [comment]
Probably not 100% germane to this work, but I am noticing this when attempting to submit new data entries:

```TypeError: Cannot read property 'dismiss' of undefined
    at InventoryDataEntriesCtrl.js?body=1:73
    at Scope.$emit (angular.js?body=1:13029)
    at InventoryDataEntryModalCtrl.vm.closeModal (InventoryDataEntryModalCtrl.js?body=1:17)
    at angular.js?body=1:10904
    at callback (angular.js?body=1:19260)
    at Scope.$eval (angular.js?body=1:12812)
    at Scope.$apply (angular.js?body=1:12910)
    at HTMLButtonElement.<anonymous> (angular.js?body=1:19265)
    at HTMLButtonElement.dispatch (jquery.js?body=1:4436)
    at HTMLButtonElement.elemData.handle (jquery.js?body=1:4122)```

File: app/assets/javascripts/client/inventories/InventoryProductEntriesCtrl.js [hunk 1]
```diff
@@ -54,8 +63,23 @@
...
+      });
+    };
+
     $scope.$on('close-product-entry-modal', function() {
+      vm.modalInstance.dismiss('cancel');
       vm.dtOptions.reloadData(null, true);
     });
+
+    function actionsHTML (data) {
+      return '' +
```
Comment (MakotoTheKnight) [inline app/assets/javascripts/client/inventories/InventoryProductEntriesCtrl.js]: No reason for us to concatenate to a blank string here.

**Comment by MakotoTheKnight** [comment]
The code here does what's advertised - it does allow me to edit entries successfully.  One papercut which is completely unrelated to this task is that the listing doesn't automatically refresh once a new entry is entered, but I think it'd be nice to put an enhancement in for that.

Other comments are as stated in the code.

**Comment by juliosantos** [comment]
> [@MakotoTheKnight] I am noticing this when attempting to submit new data entries
Actually, it's 100% germane to this work, even though it breaks nothing. Well spotted. Will fix.

**Comment by MakotoTheKnight** [comment]
LGTM (and mystically the product entries started to update; I think you were right about it being germane).

I'll be merging this in since we intend to get this deployed to staging today. :shipit:

**Pull request merged**